modules/team_planner/spec/controllers/team_planner/team_planner_controller_spec.rb

```diff
@@ -0,0 +1,45 @@
+# frozen_string_literal: true
+
+#-- copyright
+# OpenProject is an open source project management software.
+# Copyright (C) the OpenProject GmbH
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License version 3.
+#
+# OpenProject is a fork of ChiliProject, which is a fork of Redmine. The copyright follows:
+# Copyright (C) 2006-2013 Jean-Philippe Lang
+# Copyright (C) 2010-2013 the ChiliProject Team
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; either version 2
+# of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+#
+# See COPYRIGHT and LICENSE files for more details.
+#++
+
+require "spec_helper"
+require_relative "../../../../../spec/controllers/shared_upsell_examples"
+
+RSpec.describe TeamPlanner::TeamPlannerController do
+  let(:user) { create(:admin) }
+  let(:project) { create(:project) }
+
+  before do
+    login_as user
+
+    controller.instance_variable_set(:@project, project)
```

Comment:
It would at least be nice to explain why this is necessary. To me it's not _obvious_ why it's necessary to set a `@project` here. I could probably find out by seeing how exactly it fails when I remove this line, but by just scanning the code, it's not apparent to me why a `@project` might be required.